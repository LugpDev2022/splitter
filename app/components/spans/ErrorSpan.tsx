interface ErrorSpanProps {
  children: React.ReactNode;
  show?: boolean;
}

const ErrorSpan: React.FC<ErrorSpanProps> = ({ children, show = false }) => {
  return (
    <span
      className={`
      text-red-500
      font-semibold
      text-lg
      text-right
      ${show ? 'block' : 'hidden'}`}
    >
      {children}
    </span>
  );
};

export default ErrorSpan;
