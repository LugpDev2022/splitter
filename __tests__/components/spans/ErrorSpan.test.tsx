import ErrorSpan from '@/app/components/spans/ErrorSpan';
import { render, screen } from '@testing-library/react';

describe('Tests on <ErrorSpan />', () => {
  test('should match snapshot', () => {
    const { container } = render(<ErrorSpan>Test Error</ErrorSpan>);

    expect(container).toMatchSnapshot();
  });

  test('should render the given content', () => {
    const error = 'Test Error';
    render(<ErrorSpan>{error}</ErrorSpan>);

    expect(screen.getByText(error)).toBeTruthy();
  });

  test('should have hidden class', () => {
    const error = 'Test Error';
    render(<ErrorSpan show={false}>{error}</ErrorSpan>);

    const span = screen.getByText(error);
    expect(span.getAttribute('class')).toContain('hidden');
    expect(span.getAttribute('class')).not.toContain('block');
  });
});
