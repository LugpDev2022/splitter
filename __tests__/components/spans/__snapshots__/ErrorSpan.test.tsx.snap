// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests on <ErrorSpan /> should match snapshot 1`] = `
<div>
  <span
    class="
    text-red-500
    font-semibold
    text-lg
    text-right
    hidden"
  >
    Test Error
  </span>
</div>
`;
